pkg/pillar/types/global.go

```diff
@@ -689,25 +700,53 @@ func NewConfigItemSpecMap() ConfigItemSpecMap {
 	configItemSpecMap.AddIntItem(FallbackIfCloudGoneTime, 300, 60, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(MintimeUpdateSuccess, 600, 30, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(StaleConfigTime, 600, 0, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(DownloadGCTime, 600, 60, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(VdiskGCTime, 3600, 60, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(DownloadRetryTime, 600, 60, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(DomainBootRetryTime, 600, 10, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkGeoRedoTime, 3600, 60, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkGeoRetryTime, 600, 5, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkTestDuration, 30, 10, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkTestInterval, 300, 300, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkTestBetterInterval, 600, 0, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkTestTimeout, 15, 0, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(NetworkSendTimeout, 120, 0, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(Dom0MinDiskUsagePercent, 20, 20, 0xFFFFFFFF)
-	configItemSpecMap.AddIntItem(AppContainerStatsInterval, 300, 1, 0xFFFFFFFF)
+	// timer.test.baseimage.update - commit to update
+	//  This is the time to wait before declaring a baseos upgrade success.
+	configItemSpecMap.AddIntItem(MintimeUpdateSuccess, 600, 30, HourInSec)
+	// StaleConfigTime - use checkpointed config if no cloud connectivity
+	configItemSpecMap.AddIntItem(StaleConfigTime, 600, 0, DayInSec)
+	// DownloadGCTime - garbage collect unused downloaded objects
+	configItemSpecMap.AddIntItem(DownloadGCTime, 600, 60, DayInSec)
+	// VdiskGCTime - timer.gc.vdisk - garbage collect unused instance virtual disk
+	configItemSpecMap.AddIntItem(VdiskGCTime, 3600, 60, DayInSec)
+	// timer.download.retry - retry a failed download
+	configItemSpecMap.AddIntItem(DownloadRetryTime, 600, 60, DayInSec)
+	// timer.boot.retry - retry a failed domain boot
+	configItemSpecMap.AddIntItem(DomainBootRetryTime, 600, 10, DayInSec)
+	// timer.port.georedo - redo IP geolocation
+	configItemSpecMap.AddIntItem(NetworkGeoRedoTime, 3600, 60, DayInSec)
+	// timer.port.georetry - retry geolocation after failure
+	configItemSpecMap.AddIntItem(NetworkGeoRetryTime, 600, 5, DayInSec)
+	// timer.port.testduration - wait for DHCP to give address
+	configItemSpecMap.AddIntItem(NetworkTestDuration, 30, 10, DayInSec)
+	// timer.port.testinterval - retest the current port config
+	configItemSpecMap.AddIntItem(NetworkTestInterval, 300, 300, DayInSec)
+	// timer.port.testbetterinterval - test a higher prio port config
+	configItemSpecMap.AddIntItem(NetworkTestBetterInterval, 600, 0, DayInSec)
+	// timer.port.timeout - time for each http/send
+	configItemSpecMap.AddIntItem(NetworkTestTimeout, 15, 0, DayInSec)
+	// timer.send.timeout -time for each http/send
+	configItemSpecMap.AddIntItem(NetworkSendTimeout, 120, 0, DayInSec)
+	// storage.dom0.disk.minusage.percent - min. percent of persist partition
```

Comment:
Why are these arbitrary defaults any better than leaving at infinity? The fact that they exists might lead someone to believe things will work with that max setting, and that is clearly not the case. As one out of ten examples, NetworkSendTimeout being a day is utterly insane.

Also, are the comment strings consistent with the documentation in CONFIG-PROPERTIES.md? Do we need to duplicate the documentation here?

I'd drop all of these changes and move on

Reply:
> Why are these arbitrary defaults any better than leaving at infinity? The fact that they exists might lead someone to believe things will work with that max setting, and that is clearly not the case. As one out of ten examples, NetworkSendTimeout being a day is utterly insane.
> 
> Also, are the comment strings consistent with the documentation in CONFIG-PROPERTIES.md? Do we need to duplicate the documentation here?
> 
> I'd drop all of these changes and move on

Having more reasonable max values gives you visual clues on what the user should configure these values to than just using max int. And no user who looks at the code thinks about testing based on how these values are set - the code is either tested or not. For any user who looks at it, when he sees 0xFFFFFFFF as max, it is just fair game to use the value. At this point, we are just waiting for some one to actually configure that bad value, complain loudly about it - and then we come and fix it rather than start off with reasonable values to begin with .

Anyways - I don't think I can convince you nor have the energy to prolong this conversation.. Will just revert all these changes back..

Reply:
> Having more reasonable max values gives you visual clues on what the user should configure these values to than just using max int. And no user who looks at the code thinks about testing based on how these values are set - the code is either tested or not. For any user who looks at it, when he sees 0xFFFFFFFF as max, it is just fair game to use the value. At this point, we are just waiting for some one to actually configure that bad value, complain loudly about it - and then we come and fix it rather than start off with reasonable values to begin with .

Perhaps I need to explain myself better?
The arbitrary values you picked are not reasonable and either they will not work correctly in isolation or they will not work correctly in combination with other values being set to their minimums and maximums. 

Thus prentending that they might work is misleading. Users are less likely to think that infinite (4 billion) is a reasonable value and try to use it, than think that 1 day is a reasonable value and set it to that.